### PR TITLE
Implement flyin donut pause button

### LINKED ISSUE #321: Flyin Donut: Pause button
With the PR for Tiltrads' pause button, that just leaves Flyin' Donut as the only game that could make use of a pause button but currently doesn't have one

### PR DESCRIPTION
**Pull request by Brycey92**
Closes #321 

The only unintended alteration to behavior here is that, upon getting 100% beans and then pausing, the crazy colors for `100% 100% 100%` continue during pause, until you resume and the timer expires. If this is unwanted, I can go back in and figure out the math to avoid this.

### CODE REVIEW

**Comment by AEFeinstein** [comment]
Seems OK to be, but I'll wait on a review from @cnlohr since it's his mode.

**Comment by Brycey92** [comment]
I need to go re-play the 1.0 flyin donut to see how the win screen looks like it was supposed to be laid out, so I can get the x offsets programmatically. Currently, everything on the screen is centered except the time.

**Comment by Brycey92** [comment]
It was a little messy in the original, sort of left-aligned with each other, sort-of centered. Standardizing on centered unless anyone has objections.

**Comment by AEFeinstein** [comment]
No objections

**Review by cnlohr — APPROVED** [review]

**Pull request merged**
